src/Meshes/BoxMesh.jl

```diff
@@ -1,3 +1,16 @@
+function equispaced_rectangular_mesh(
```

Comment:
How is this different from the existing [`EquispacedRectangleMesh`](https://github.com/CliMA/ClimaCore.jl/blob/0375673a843f263f9a5186db886e6cc32cc8b068/src/Meshes/Meshes.jl#L110). If it is not that different, then why not simply use the existing one?